src/Library/helpers.php

```diff
@@ -1,17 +1,38 @@
 <?php
 
+use Illuminate\Support\Collection;
+use Illuminate\Pagination\LengthAwarePaginator;
 use OwowAgency\LaravelResources\Resources\ResourceFactory;
 
 if (! function_exists('resource')) {
     /**
      * Makes the resource for the specified model class.
      *
      * @param  mixed  $model
-     * @param  boolean  $isCollection
+     * @param  boolean  $isCollection|null
      * @return mixed
      */
-    function resource($model, $isCollection = false)
+    function resource($model, $isCollection = null)
     {
-        return (new ResourceFactory)->make($model, $isCollection);
+        if (is_null($isCollection)) {
+            if (
+                $model instanceof Collection
+                || $model instanceof LengthAwarePaginator
+            ) {
+                $isCollection = true;
+            } else {
+                $isCollection = false;
+            }
+        }
```

Comment:
Can be done simpler
```suggestion
        $isCollection = $isCollection
            ?? ($model instanceof Collection || $model instanceof LengthAwarePaginator)
```
Not sure if those parentesis are required.